pydm/widgets/waveformplot.py

```diff
@@ -5,85 +5,204 @@
+	
+	@property
+	def y_address(self):
+		if self.y_channel is None:
+			return None
+		return self.y_channel.address
+	
+	@y_address.setter
+	def y_address(self, new_address):
+		if new_address is None or len(str(new_address)) < 1:
+			self.y_channel = None
+			return
+		self.y_channel = PyDMChannel(address=new_address, connection_slot=self.yConnectionStateChanged, waveform_slot=self.receiveYWaveform)
+	
+	def to_dict(self):
+		return OrderedDict([("y_channel", self.y_address), ("x_channel", self.x_address), ("name", self.name()), ("color", self.color_string)])
+	
+	@pyqtSlot(bool)
+	def xConnectionStateChanged(self, connected):
+		pass
+		
+	@pyqtSlot(bool)
+	def yConnectionStateChanged(self, connected):
+		pass
+	
+	@pyqtSlot(np.ndarray)
+	def receiveXWaveform(self, new_waveform):
+		self.x_waveform = new_waveform
+		#Don't redraw unless we already have Y data.
+		if self.y_waveform is not None:
+			self.data_changed.emit()
+	
+	@pyqtSlot(np.ndarray)
+	def receiveYWaveform(self, new_waveform):
+		self.y_waveform = new_waveform
+		if self.x_channel is None or self.x_waveform is not None:
+			self.data_changed.emit()
+	
+	def redrawCurve(self):
+		#We try to be nice: if the X waveform doesn't have the same number of points as the Y waveform,
+		#we'll truncate whichever was longer so that they are both the same size.
+		if self.x_waveform is None:
+			self.setData(y=self.y_waveform)
+			return
+		if self.x_waveform.shape[0] > self.y_waveform.shape[0]:
+				self.x_waveform = self.x_waveform[:self.y_waveform.shape[0]]
+		elif self.x_waveform.shape[0] < self.y_waveform.shape[0]:
+				self.y_waveform = self.y_waveform[:self.x_waveform.shape[0]]
+		self.setData(x=self.x_waveform, y=self.y_waveform)
+	
+	def limits(self):
+		"""Returns a nested tuple of limits: ((xmin, xmax), (ymin, ymax))"""
+		if self.y_waveform is None:
+			raise NoDataError("Curve has no Y data, cannot determine limits.")
+		if self.x_waveform is None:
+			yspan = float(np.amax(self.y_waveform)) - float(np.amin(self.y_waveform))
+			return ((0, len(self.y_waveform)), (float(np.amin(self.y_waveform) - yspan), float(np.amax(self.y_waveform) + yspan)))
+		else:
+			return ((np.amin(self.x_waveform), np.amax(self.x_waveform)), (np.amin(self.y_waveform), np.amax(self.y_waveform)))
+	
+class PyDMWaveformPlot(BasePlot):
+	def __init__(self, parent=None, init_x_channels=[], init_y_channels=[], background='default'):
+		super(PyDMWaveformPlot, self).__init__(parent, background)
+		#If the user supplies a single string instead of a list, wrap it in a list.
+		if isinstance(init_x_channels, str):
+			init_x_channels = [init_x_channels]
+		if isinstance(init_y_channels, str):
+			init_y_channels = [init_y_channels]
+		if len(init_x_channels) == 0:
+			init_x_channels = list(itertools.repeat(None, len(init_y_channels)))
+		if len(init_x_channels) > 0 and len(init_x_channels) != len(init_y_channels):
```

Comment:
Yep!  That check was left over from an old version.  It is always satisfied.  I'll remove it.